File: examples/account_info/account_info.cs
using System;

namespace account_info_cs
{
    class account_info
    {

        static void Main(string[] args)
        {

            using (AergoClient client = new AergoClient())
            {
                client.Connect("testnet-api.aergo.io", 7845);

                AergoClient.AergoAccount account = new AergoClient.AergoAccount
                {
                    privkey = new byte[] {
                        0xDB, 0x85, 0xDD, 0x0C, 0xBA, 0x47, 0x32, 0xA1, 0x1A, 0xEB,
                        0x3C, 0x7C, 0x48, 0x91, 0xFB, 0xD2, 0xFE, 0xC4, 0x5F, 0xC7,
                        0x2D, 0xB3, 0x3F, 0xB6, 0x1F, 0x31, 0xEB, 0x57, 0xE7, 0x24,
                        0x61, 0x76
                    }
                };

                Console.WriteLine("Sending request...");

                var ret = client.GetAccountInfo(ref account);

                if (ret.success) {
                    Console.WriteLine("Account state:");
                    Console.WriteLine("Address = " + account.address);
                    Console.WriteLine("Balance = " + account.balance);
                    Console.WriteLine("Nonce   = " + account.nonce);
                } else {
                    Console.WriteLine("Failed: " + ret.error);
                }

            }

            Console.WriteLine("Press any key to continue");
            Console.ReadKey();

        }        
    }
}

File: examples/account_info/account_info.vb
Module account_info_example

    Sub Main()

        Using client As New AergoClient()

            client.Connect("testnet-api.aergo.io", 7845)

            Dim account As AergoClient.AergoAccount = New AergoClient.AergoAccount With
                {.privkey = New Byte() {&HCD, &HFA, &H3B, &HF9, &H1A, &H2C, &HB6, &H9B, &HEC, &HB0,
                                         &H16, &H7E, &H11, &H0, &HF6, &HDE, &H77, &HAF, &H5, &HD6,
                                         &H4E, &HE5, &HC, &H2C, &HD4, &HBA, &HDD, &H70, &H1, &HF0,
                                         &HC5, &H4B}}

            ' or use an account on Ledger Nano S
            'Dim account As AergoClient.AergoAccount = New AergoClient.AergoAccount With
            '    {.use_ledger = True}

            Dim ret = client.GetAccountInfo(account)

            If ret.success Then
                Console.WriteLine("Account state:")
                Console.WriteLine("Address = " + account.address)
                Console.WriteLine("Balance = " + account.balance.ToString())
                Console.WriteLine("Nonce   = " + account.nonce.ToString())
            Else
                Console.WriteLine("FAILED: " + ret.error)
            End If

        End Using

        Console.WriteLine("Press any key to continue")
        Console.ReadKey()

    End Sub

End Module
